localization/CMakeLists.txt

```diff
@@ -3,6 +3,9 @@ add_custom_target(diagnostic-database)
 add_custom_command(TARGET diagnostic-database
   COMMAND
     ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/diagnostics/ ${CMAKE_BINARY_DIR}/share/swift/diagnostics/
+  COMMAND
+    "${SWIFT_NATIVE_SWIFT_TOOLS_PATH}/swift-def-to-yaml-converter"
```

Comment:
Always converting might be too time consuming. Could it be done only if there is no `.yaml` and/or `.def` has been changed between the builds?

Reply:
I'm not sure how can I do that, I tried using DEPENDS and gave it the path to a .def file and didn't modify it, but it created a new file anyways...

Reply:
@compnerd Do you know if there a way to check whether a particular file has been modified `DiagnosticsSema.def` (for example)? @HassanElDesouky How long does it take to convert `DiagnosticsAll.def` into yaml?

Reply:
an average of 15 to 20 ms.

Reply:
@xedin Maybe `add_custom_command` [`DEPENDS`](https://cmake.org/cmake/help/v3.13/command/add_custom_command.html) can do the trick here like `DEPENDS ${PATH_TO_DEF_FILE}` not really sure but I think that could work to only run if `.def` changes

Reply:
`DEPENDS` ensures existence of .def files, it would be more useful to check whether it's changed, but I think it's okay since conversion doesn't take too much time...

Reply:
ahh, I thought that DEPENDS could do it for change files as well, but you right. I was just curious to know if cmake has something built-in for this on custom command, but couldn't find anything yet :(